aries_cloudagent/revocation/recover.py

```diff
@@ -0,0 +1,113 @@
+"""Recover a revocation registry."""
+
+import hashlib
+import importlib
+import logging
+import tempfile
+import time
+
+import aiohttp
+import base58
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+"""
+This module calculates a new ledger accumulator, based on the revocation status
+on the ledger vs revocations recorded in the wallet.
+The calculated transaction can be written to the ledger to get the ledger back
+in sync with the wallet.
+This function can be used if there were previous revocation errors (i.e. the
+credential revocation was successfully written to the wallet but the ledger write
+failed.)
+"""
```

Comment:
I tried to keep this script relatively self-contained (no dependencies on other aca-py code) to make it easier to port if anyone else needs to do this (any other agents), I expect the revocation corruption problem will be fairly common.